conmon/conmon.c

```diff
@@ -632,6 +636,45 @@ int main(int argc, char *argv[])
 	cpid = atoi(contents);
 	ninfo("container PID: %d", cpid);
 
+	/* Setup endpoint for attach */
+	struct sockaddr_un attach_addr = {0};
+	_cleanup_close_ int afd = -1;
+	int asfd = -1;
+
+	if (!exec) {
+		attach_addr.sun_family = AF_UNIX;
+		snprintf(attach_sock_path, PATH_MAX, "/var/run/%s-attach", cid);
+		ninfo("attach sock path: %s", attach_sock_path);
+
+		asfd = open(attach_sock_path, O_CREAT|O_WRONLY);
+		if (asfd == -1) {
+			pexit("Failed to create attach socket file");
+		}
+		close(asfd);
+
+		strncpy(attach_addr.sun_path, attach_sock_path, sizeof(attach_addr.sun_path) - 1);
+		ninfo("addr{sun_family=AF_UNIX, sun_path=%s}", attach_addr.sun_path);
+
+		/*
+		 * We make the socket non-blocking to avoid a race where client aborts connection
+		 * before the server gets a chance to call accept. In that scenario, the server
+		 * accept blocks till a new client connection comes in.
+		 */
+		afd = socket(AF_UNIX, SOCK_STREAM | SOCK_NONBLOCK, 0);
+		if (afd == -1)
+			pexit("Failed to create attach socket");
+
+		if (unlink(attach_sock_path) == -1 && errno != ENOENT) {
+			pexit("Failed to remove socket file");
+		}
+
```

Comment:
Might want to call fchmod() here before bind to make the new socket mode 0600 or something.